packages/hdwallet-provider/src/index.ts

```diff
@@ -215,18 +217,97 @@ class HDWalletProvider {
     });
   }
 
-  public send(
+  private initialize(): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this.engine.sendAsync({
+        jsonrpc: '2.0',
+        id: Date.now(),
+        method: 'eth_chainId',
+        params: []
+    }, (error: any, response: JSONRPCResponsePayload) => {
+        if (error) {
+          reject(error);
+        }
+        if (response && response.result) {
+          if (isNaN(parseInt(response.result, 16))) {
+            const message = "When requesting the chain id from the node, it" +
+              `returned the malformed result ${response.result}.`;
+            throw new Error(message);
+          }
+          this.chainId = response.result;
+        }
+        resolve();
+      });
+    });
+  }
+
+  // private helper to check if given mnemonic uses BIP39 passphrase protection
+  private checkBIP39Mnemonic ({
+    addressIndex,
+    numberOfAddresses,
+    phrase,
+    password
+  }: {
+    addressIndex: number;
+    numberOfAddresses: number;
+    phrase: string;
+    password?: string;
+  }) {
+    this.hdwallet = EthereumHDKey.fromMasterSeed(
+      bip39.mnemonicToSeedSync(phrase, password)
+    );
+
+    if (!bip39.validateMnemonic(phrase, wordlist)) {
+      throw new Error("Mnemonic invalid or undefined");
+    }
+
+    // crank the addresses out
+    for (let i = addressIndex; i < addressIndex + numberOfAddresses; i++) {
+      const wallet = this.hdwallet
+        .derivePath(this.walletHdpath + i)
+        .getWallet();
+      const addr = `0x${wallet.getAddress().toString("hex")}`;
+      this.addresses.push(addr);
+      this.wallets[addr] = wallet;
+    }
+  }
+
+  // private helper leveraging ethUtils to populate wallets/addresses
+  private ethUtilValidation ({
+    addressIndex,
+    privateKeys
+  }: {
+    addressIndex: number;
+    privateKeys: string[];
+  }) {
+    // crank the addresses out
+    for (let i = addressIndex; i < privateKeys.length; i++) {
+      const privateKey = Buffer.from(privateKeys[i].replace("0x", ""), "hex");
+      if (EthUtil.isValidPrivate(privateKey)) {
+        const wallet = ethJSWallet.fromPrivateKey(privateKey);
+        const address = wallet.getAddressString();
+        this.addresses.push(address);
+        this.wallets[address] = wallet;
+      }
+    }
+  };
+
+
+  public async send(
```

Comment:
My gut is telling me you'll want to keep this as `non-async` and just use the `.then` pattern you used for `sendAsync` here as well.

Reply:
So I did this because we were previously returning a value. Do we just want to fire this one off async and return the Promise? And in that case, would it be any different than what I've done here?

Reply:
Ah. Looks like Web3ProviderEngine throws on send anyway? https://github.com/MetaMask/web3-provider-engine/blob/2189a1a74552cf88cff6b1ec2b61aac6769fc92e/index.js#L114

Have you tried this?

Reply:
Answered in Slack but leaving here for reference; Truffle uses an old, forked version of web3-provider-engine which calls `sendAsync` from `send`. https://github.com/trufflesuite/provider-engine/blob/master/index.js#L116-L118